testers.testVersion: move from trivial-builders.nix

Artturin opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Artturin committed:
testers.testVersion: move from trivial-builders.nix

we will have more testers in the future so they should have their own
location

putting 'testers' in args will also make it simpler to use multiple testers

Artturin committed:
treewide: testVersion -> testers.testVersion

Artturin commented:
added an alias

FRidh reviewed (dismissed):
Main commit lacks a motivation. What is behind this? A new structure? Should we get docs for these in the manual?

Artturin committed:
testers: convert to a format that is kind of compatible with nixdoc

Examples: support has to be added to https://github.com/nix-community/nixdoc

'nixdoc --category testers --description "nixpkgs testers" --file ./pkgs/build-support/testers/default.nix'

Artturin commented:
> Main commit lacks a motivation. What is behind this? A new structure? Should we get docs for these in the manual?

motivation added, getting docs either needs a update to nixdoc or moving the docs to somewhere in `./doc/builders/special/invalidateFetcherByDrvHash.section.md`  https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-invalidateFetcherByDrvHash

Artturin committed:
doc: move testers to their own chapter

Artturin commented:
Ready to merge

I'll make another pr that moves the invalidate @roberth

roberth commented on pkgs/top-level/aliases.nix:
```diff
@@ -1253,6 +1253,7 @@ mapAliases ({
   terraform_1_0 = throw "terraform_1_0 has been renamed to terraform_1"; # Added 2021-12-08
   terraform_1_0_0 = throw "terraform_1_0_0 has been renamed to terraform_1"; # Added 2021-06-15
   tesseract_4 = throw "'tesseract_4' has been renamed to/replaced by 'tesseract4'"; # Converted to throw 2022-02-22
+  testVersion = throw "'testVersion' has been moved to 'testers.testVersion'"; # Added 2022-04-20
```
Let's make this a proper alias, so that out-of-tree expressions can use the old location until all nixpkgs versions without the new location are end of life.

I've set up a little bit of code and process to support this. `lib.isInOldestRelease 2205` will evaluate to `true` when the time to start the migration comes. https://github.com/NixOS/release-wiki/pull/32 It's a new thing, so might need some clarification and some helper functions and whatnot.

Artturin replied:
changed it to a alias

roberth commented on pkgs/tools/misc/arch-install-scripts/default.nix:
```diff
@@ -6,7 +6,7 @@
 , gawk
 , gnum4
-, testVersion
+, testers
```
Unused.

roberth reviewed (commented):
Just the one thing. Otherwise lgtm.

bobby285271 commented on doc/builders/testers.chapter.md:
````diff
@@ -0,0 +1,82 @@
+Examples:
+
+```nix
+passthru.tests.version = testVersion { package = hello; };
````
Maybe `passthru.tests.version = testers.testVersion { package = hello; };`?
